### PR TITLE
Foundational Training Framework & Experiments

### PR DESCRIPTION
**Pull request by JakeColor**
Introduces foundational training framework for downscaling with CNN's.

Implements several variations of CNNs tested in Bano Medina's 2020 paper, and scales up the best performer.

### CODE REVIEW

**Review by JakeColor — COMMENTED** [review]
SRC

File: src/training/data.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self.batch_size = batch_size
+
+        if not (self.coarse.index.levels[0] == self.fine.index.levels[0]).all():
+            raise ValueError("coarse-grid dates do not match fine-grid dates")
+
+    def setup(self, stage):
+        """ Determines transformation, splits, etc. """
+        # demean
+        daily_means = self.coarse.groupby("date").mean()
+        coarse_demeaned = self.coarse - daily_means
```
Comment (JakeColor) [inline src/training/data.py]: we also need to divide by std here

**Review by JakeColor — COMMENTED** [review]
SRC

**Review by hemanthsheetha — APPROVED** [review]

File: src/data/standardization.ipynb [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "list(random_dates)"
+   ]
+  },
+
```
Comment (oethier) [inline src/data/standardization.ipynb:192]: Here, I think I would change the vmin and vmax for the standardized plot to something like -5, 5. That way, we could visually compare before and after standardization. Right now, the second image is mostly just white.

Comment (JakeColor) [reply src/data/standardization.ipynb:192]: that's the point! the variance of the training data is much less than it was before

**Pull request merged**